examples/examples/jsonrpsee_server_low_level_api.rs

```diff
@@ -89,14 +90,47 @@ where
 
 			if *lock >= 10 {
 				let _ = state.try_send(());
-				MethodResponse::error(req.id, ErrorObject::borrowed(-32000, "RPC rate limit", None))
+				Ok(MethodResponse::error(req.id, ErrorObject::borrowed(-32000, "RPC rate limit", None)))
 			} else {
 				let rp = service.call(req).await;
 				*lock += 1;
 				rp
 			}
 		}
-		.boxed()
+	}
+
+	fn batch<'a>(&self, batch: Batch<'a>) -> impl Future<Output = Result<Self::Response, Self::Error>> + Send + 'a {
+		let count = self.count.clone();
+		let state = self.state.clone();
+		let service = self.service.clone();
+
+		async move {
+			let mut lock = count.lock().await;
+
+			if *lock >= 10 {
+				let _ = state.try_send(());
+				Ok(MethodResponse::error(Id::Null, ErrorObject::borrowed(-32000, "RPC rate limit", None)))
+			} else {
+				let batch_len = batch.len();
+				let rp = service.batch(batch).await;
+				*lock += batch_len;
```

Comment:
What if the batch len is like 100?

Would it make sense to do something like "if *lock + batch.len() >= 10"?

Reply:
yupp :)